### ISSUE TITLE
web-speech-cognitive-services dependency is missing

### ISSUE THREAD

**Issue by johnataylor** [microsoft/BotFramework-WebChat#3042]
When building with the latest **4.8.0** package the following error is produced:

```
ERROR in ./node_modules/botframework-webchat/lib/createCognitiveServicesBingSpeechPonyfillFactory.js
Module not found: Error: Can't resolve 'web-speech-cognitive-services/lib/BingSpeech' in 'C:\private\v4\BotBuilder-Samples\samples\javascript_es6\01.browser-echo\node_modules\botframework-webchat\lib'
 @ ./node_modules/botframework-webchat/lib/createCognitiveServicesBingSpeechPonyfillFactory.js 26:41-96
 @ ./node_modules/botframework-webchat/lib/index.js
 @ ./src/app.ts
```

Attempting a work around by explicitly adding the web-speech-cognitive-services/6.0.0 package to the application fixes the build issue but now the runtime fails. This is just seen as a blank screen in the brower-echo sample.

(Note in the process of updating the sample now you have to also add react and react-dom as explicit dependencies - this is intentional as it allows the application to choose which version of react it wants to use.)

**Comment by zp1112** [comment]
I had the same problem,how to fix it?

**Comment by yxc2010** [comment]
https://github.com/compulim/web-speech-cognitive-services/commit/d6224a4501286913a831f4edf17afd03c5c32699

### Breaking changes

- As Bing Speech has deprecated and removed from Azure. Bing Speech support has been removed.

I downgraded to BotFramework-WebChat@4.6.0 for addressing this issue.

**Comment by zp1112** [comment]
> [compulim/web-speech-cognitive-services@d6224a4](https://github.com/compulim/web-speech-cognitive-services/commit/d6224a4501286913a831f4edf17afd03c5c32699)
> 
> ### Breaking changes
> * As Bing Speech has deprecated and removed from Azure. Bing Speech support has been removed.
> 
> I downgraded to BotFramework-WebChat@4.6.0 for addressing this issue.
> 
> 4.8.0 can't be built now.

installed 4.6.0,but throw error hooks is undefined

**Comment by Jakob-vdH** [comment]
> [compulim/web-speech-cognitive-services@d6224a4](https://github.com/compulim/web-speech-cognitive-services/commit/d6224a4501286913a831f4edf17afd03c5c32699)
> 
> ### Breaking changes
> * As Bing Speech has deprecated and removed from Azure. Bing Speech support has been removed.
> 
> I downgraded to BotFramework-WebChat@4.6.0 for addressing this issue.
> 
> 4.8.0 can't be built now.

downgrading to v4.6.0 worked, but results in for example suggested actions not being displayed stacked anymore, which we require. 

Removing `web-speech-cognitive-services/lib/BingSpeech` and it's references from the botframework-webchat package, led to a successfull build, but also to further errors at runtime with the useDateFromatter hook in the StackedLayout of the chat `botframework-webchat-component/lib/Activity/StackedLayout` :
```
Uncaught TypeError: numberToPartsFormatter is not a function
    at Object.numberFormatter (number.js:505)
    at date.js:636
    at String.replace (<anonymous>)
    at dateFormat (date.js:345)
    at dateToPartsFormatter (date.js:1462)
    at dateFormatter (date.js:648)
    at useDateFormatter.js:69
    at StackedLayout (StackedLayout.js:205)
```

**Comment by zp1112** [comment]
> web-speech-cognitive-services

where to remove web-speech-cognitive-services?could you please show the detail?thank you very much

**Comment by Jakob-vdH** [comment]
> > web-speech-cognitive-services
> 
> where to remove web-speech-cognitive-services?could you please show the detail?thank you very much

sure.

- `botframework-webchat\lib\creteCognitiveServicesBingSpeechPonyFillFactory.js` - line 26 
(`var _BingSpeech = _interopRequireDefault(require("web-speech-cognitive-services/lib/BingSpeech"));`)

- `botframework-webchat\lib\index.js` - line 26-31, 79, 132 all of them deal with `_createCognitiveServicesBingSpeechPonyfillFactory`

This is only a temporary workaround, but allowed us to proceed with our current development.
However, you will not have the features of v4.8.0 this way.

**Comment by zp1112** [comment]
> > > web-speech-cognitive-services
> > 
> > 
> > where to remove web-speech-cognitive-services?could you please show the detail?thank you very much
> 
> sure.
> 
> * `botframework-webchat\lib\creteCognitiveServicesBingSpeechPonyFillFactory.js` - line 26
>   (`var _BingSpeech = _interopRequireDefault(require("web-speech-cognitive-services/lib/BingSpeech"));`)
> * `botframework-webchat\lib\index.js` - line 26-31, 79, 132 all of them deal with `_createCognitiveServicesBingSpeechPonyfillFactory`
> 
> This is only a temporary workaround, but allowed us to proceed with our current development.
> However, you will not have the features of v4.8.0 this way.

This can fix development mode,but how to workaround production mode?

**Comment by Jakob-vdH** [comment]
> This can fix development mode,but how to workaround production mode?

My first thought would be to build your own private package of the BotFramework-WebChat ([CONTRIBUTING.md](https://github.com/microsoft/BotFramework-WebChat/blob/master/.github/CONTRIBUTING.md)).

**Comment by sgellock** [comment]
The team is working through how to get a fix out to address this.  We are in a release lockdown due to current world events.  but we're working to see if we can get an exemption and get out a fix.  we'll post more data here as we gather it.

**Comment by cwhitten** [comment]
Hi all.

We're actively working on addressing this and hope to have an update in a couple hours. I'll update this issue when we have more information. 

Thanks!

**Comment by compulim** [comment]
I am verifying the fix right now.

**Comment by compulim** [comment]
We are still working on a fix.

For temporarily workaround, please install a local version of `web-speech-cognitive-services@6.0.0` so Web Chat 4.8.0 will pick up the older release.

Also, we discovered jQuery's `globalize@1.5.0` introduced a bug and causing JavaScript errors. We did not see the problem on `globalize@1.4.2` (which we tested) and `globalize@1.4.3`.

So, for workaround, after you install Web Chat, run the following:

```sh
npm install web-speech-cognitive-services@6.0.0 globalize@1.4.2 --save
```

Since Web Chat relies on `@^6.0.0`, it will use the local version of `@6.0.0` and not installing the incompatible `@6.1.0`.

**Comment by compulim** [comment]
This is fixed in `web-speech-cognitive-services@6.3.0`, which should be automatically picked up by reinstalling `botframework-webchat@4.8.0`.

In case it is not, please check your `package-lock.json` to make sure it doesn't lock down on `@6.1.0-6.2.0`.

But as @Jakob-vdH pointed out, `globalize` also recently released a package that broke semver rule. It is being reported at https://github.com/globalizejs/globalize/issues/895 and #3046. The temporary workaround will be downgrading `globalize` to `1.4.2`.

Please refer to #3046 for status and instructions.
